🌐 Add Spanish translation for `docs/es/docs/contributing.md` (#10965)
fastapi/fastapi: docs/es/docs/contributing.md

### Instalar requerimientos utilizando pip

Después de activar el entorno como se describe arriba:

<div class="termy">

```console
$ pip install -r requirements.txt

---> 100%
```

</div>

Esto instalará todas las dependencias y tu FastAPI local en tu entorno local.

### Utilizando tu FastAPI local

Si creas un archivo Python que importa y usa FastAPI, y lo ejecutas con Python desde tu entorno local, usará tu código fuente local clonado de FastAPI.

Y si actualizas ese código fuente local de FastAPI cuando vuelvas a ejecutar ese archivo Python, utilizará la versión nueva de FastAPI que acabas de editar.

De esa manera, no tendrás que "instalar" tu versión local para poder probar cada cambio.

!!! note "Detalles Técnicos"
    Esto solo sucede cuando realizas la instalación utilizando el `requirements.txt` incluido en lugar de ejecutar `pip install fastapi` directamente.

    Esto se debe a que dentro del archivo `requirements.txt`, la versión local de FastAPI está marcada para instalarse en modo "editable", con la opción `-e`.

### Formatear el código

> Hay un script que puedes ejecutar y que formateará y limpiará todo tu código:

Reviewer: There is an extra`y` (the first one) that is confusing when reading
```suggestion
Hay un script que puedes ejecutar que formateará y limpiará todo tu código:
```

<div class="termy">

```console
$ bash scripts/format.sh
```

</div>

También ordenará automáticamente todas tus importaciones.

> Para ordenarlos correctamente, necesitas tener FastAPI instalado localmente en tu entorno, con el comando de la sección anterior usando `-e`.

Reviewer: `importaciones` is feminine, also I would use `utilizando`to be clearer
```suggestion
Para ordenarlas correctamente, necesitas tener FastAPI instalado localmente en tu entorno, utilizando el comando de la sección anterior usando `-e`.
```

## Documentación

> Primero, asegúrese de configurar su entorno como se describe anteriormente, eso instalará todos los requisitos.

Reviewer: Previously you use `requerimientos` for `requirements` ([here](https://github.com/tiangolo/fastapi/pull/10965/files#diff-95a846f155a6570f1c255067c29ccdbd0a784effd0d46d318de93dad472301c6R104)). To be consistent, we need to use the same translation here.

```suggestion
Primero, asegúrese de configurar su entorno como se describe anteriormente, eso instalará todos los requerimientos.
```

### Documentación en vivo

Durante el desarrollo local, hay un script que construye el sitio y verifica si hay cambios, recargando en vivo:

<div class="termy">

```console
$ python ./scripts/docs.py live

<span style="color: green;">[INFO]</span> Serving on http://127.0.0.1:8008
<span style="color: green;">[INFO]</span> Start watching changes
<span style="color: green;">[INFO]</span> Start detecting changes
```

</div>

Servirá la documentación en `http://127.0.0.1:8008`.

De esa manera, puedes editar los archivos de documentación/código fuente y ver los cambios en vivo.

!!! tip "Consejo"
    Alternativamente, puedes realizar los mismos pasos que los scripts realizan manualmente.

    Ve al directorio de idiomas, para los documentos principales en inglés en `docs/en/`:

    ```console
    $ cd docs/en/
    ```

    Luego ejecuta `mkdocs` en ese directorio:

    ```console
    $ mkdocs serve --dev-addr 8008
    ```

#### Typer CLI (opcional)

Las instrucciones aquí te muestran cómo utilizar el script en `./scripts/docs.py` directamente con el programa `python`.

> Pero también puedes usar <a href="https://typer.tiangolo.com/typer-cli/" class="external-link" target="_blank">Typer CLI</a> y obtendrás el autocompletado en tu terminal para ver los comandos después de instalar el completamiento.

Reviewer: It is not a real word `completamiento`
```suggestion
Pero también puedes usar <a href="https://typer.tiangolo.com/typer-cli/" class="external-link" target="_blank">Typer CLI</a> y obtendrás el autocompletado en tu terminal para ver los comandos después de instalar el completado.
```

Author: Puedo cambiarlo... pero la palabra si existe en español:

https://dle.rae.es/completamiento

Reviewer: ahh, disculpas entoncés. Tendría que haberlo buscado ahi tambien. Si preferís que quede como antes por mi OK

> Si instalas Typer CLI, puedes instalar el completamiento con:

Reviewer: ```suggestion
Si instalas Typer CLI, puedes instalar el completado con:
```

Author: Puedo cambiarlo... pero la palabra si existe en español:

https://dle.rae.es/completamiento

<div class="termy">

```console
$ typer --install-completion

zsh completion installed in /home/user/.bashrc.
Completion will take effect once you restart the terminal.
```

</div>

### Estructura de la Documentación

La documentación utiliza <a href="https://www.mkdocs.org/" class="external-link" target="_blank">MkDocs</a>.

Y hay herramientas/scripts adicionales para manejar las traducciones en `./scripts/docs.py`.

!!! tip "Consejo"
    No necesitas ver el código en `./scripts/docs.py`, solo úsalo en la línea de comando.

Toda la documentación está en formato Markdown en el directorio. `./docs/en/`.

Muchos de los tutoriales tienen bloques de código.

> En la mayoría de los casos, estos bloques de código son aplicaciones reales completas que se pueden ejecutar tal cual.

Reviewer: I would add `están` at the end, i think is more idiomatic
```suggestion
En la mayoría de los casos, estos bloques de código son aplicaciones reales completas que se pueden ejecutar tal cual están.
```

> De hecho, esos bloques de código no están escritos dentro de Markdown, son archivos de Python en el directorio `./docs_src/`.

Reviewer: ```suggestion
De hecho, esos bloques de código no están escritos dentro del Markdown, son archivos de Python en el directorio `./docs_src/`.
```

Author: No entiendo el cambio. En el contexto se refiere a que los bloques de códigos no están escritos dentro de Markdown. Por ejemplo: Esta documentación no está escrita dentro de Python. En estos casos sonaría un poco raro el uso de: "del".

Por el mometo no actualizaré esta sugerencia en espera de su respuesta. 

Gracias

Y esos archivos Python se incluyen/inyectan en la documentación cuando se genera el sitio.

### Documentación para las pruebas

La mayoría de las pruebas en realidad se ejecutan con los archivos de código fuente de ejemplo en la documentación.

Esto ayuda a garantizar que:

> *La documentación está actualizada.

Reviewer: missed space 
<img width="768" alt="image" src="https://github.com/tiangolo/fastapi/assets/65260383/f15c4d3d-7b35-4862-a6b3-32fa81e12f01">

```suggestion
* La documentación está actualizada.
```

> * Los ejemplos de documentación se pueden ejecutar tal cual.

Reviewer: ```suggestion
* Los ejemplos de documentación se pueden ejecutar tal cual  están.
```

> * La mayoría de las características están cubiertas por la documentación, garantizado por la cobertura de las pruebas.

Reviewer: ```suggestion
* La mayoría de las características están cubiertas por la documentación, aseguradas por la cobertura de las pruebas.
```

#### Aplicaciones y documentación al mismo tiempo

Si ejecuta los ejemplos con, ej.:

<div class="termy">

```console
$ uvicorn tutorial001:app --reload

<span style="color: green;">INFO</span>:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
```

</div>

como Uvicorn utilizará por defecto el puerto `8000`, la documentación sobre el puerto `8008` no entrará en conflicto.

### Traducciones

¡Se agradece MUCHO la ayuda con las traducciones! Y no se puede hacer sin la ayuda de la comunidad. 🌎 🚀

Estos son los pasos para ayudar con las traducciones.

#### Consejos y recomendaciones

> * Verifica los <a href="https://github.com/tiangolo/fastapi/pulls" class="external-link" target="_blank">pull requests existentes</a> actualmente para tu idioma. Puedes filtrar los pull requests por aquellas que tienen la etiqueta para tu idioma. Por ejemplo, para español, la etiqueta es <a href="https://github.com/tiangolo/fastapi/pulls?q=is%3Aopen+sort%3Aupdated-desc+label%3Alang-es+label%3Aawaiting- revisión" class="external-link" target="_blank">`lang-es`</a>.

Reviewer: ```suggestion
* Revisa los <a href="https://github.com/tiangolo/fastapi/pulls" class="external-link" target="_blank">pull requests existentes</a> actualmente para tu idioma. Puedes filtrar los pull requests por aquellos con la etiqueta de tu idioma. Por ejemplo, para español, la etiqueta es <a href="https://github.com/tiangolo/fastapi/pulls?q=is%3Aopen+sort%3Aupdated-desc+label%3Alang-es+label%3Aawaiting- revisión" class="external-link" target="_blank">`lang-es`</a>.
```

> * Revisa esos pull request, solicitando cambios o aprobándolos. Para los idiomas que no hablo, esperaré a que otros revisen la traducción antes de fusionarlos.

Reviewer: ```suggestion
* Revisa esos pull request, solicitando cambios o aprobándolos. Para los idiomas que no hablo, esperaré a que otros revisen la traducción antes de mergearlos.
```

Author: "mergearlos" es una palabra que no existe en el diccionario. Popularmente si se utiliza por lo que utilizaré:
<abbr title="también conocido como: mezclarlos, mergearlos">fusionarlos</abbr>
